Add plugin: Flashy

MasonGuinn opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/MasonGuinn/obsidian-flashy

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [ x My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

MasonGuinn commented:
Any update?

mnaoumov commented:
Please see below the list of issues found during the plugin review. Additionally, provided the (case-sensitive) regular expression you can use to search for all the places in your codebase to address similar issues.

## 1

[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

https://github.com/MasonGuinn/obsidian-flashy/blob/25d5ea8f5eb9c72547367c00bcb6c456bb2f22c8/main.ts#L114

**Regular expression**: `[A-Z]\w+ [A-Z]\w+`

## 2

Use `window.setTimeout`, `window.clearTimeout`, `window.setInterval`, `window.clearInterval` instead of their versions without `window.`

Use `number` instead of `NodeJs.Timeout`.

https://github.com/MasonGuinn/obsidian-flashy/blob/25d5ea8f5eb9c72547367c00bcb6c456bb2f22c8/main.ts#L207

**Regular expression**: `(set|clear)(Timeout|Interval)`

## 3

[Use setHeading instead of a \<h1\>, \<h2\>](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+%60setHeading%60+instead+of+a+%60%3Ch1%3E%60%2C+%60%3Ch2%3E%60)

https://github.com/MasonGuinn/obsidian-flashy/blob/25d5ea8f5eb9c72547367c00bcb6c456bb2f22c8/main.ts#L263

**Regular expression**: `\bh1|\bh2|\bh3`

## 4

[No hardcoded styling](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#No+hardcoded+styling)

**Regular expression**: `\bstyle\b`

MasonGuinn commented:
Hello again!

Thanks for the review. I've addressed the initial feedback in release 1.0.1.

Since then, I also fixed a bug with the card shuffling on reset and added several new settings based on user feedback (configurable keyboard shortcuts, option to disable keyboard navigation, option to auto-advance on incorrect answers). These changes, along with the original review fixes, are included in the latest release, 1.1.0.

Here's the link to the updated release: https://github.com/MasonGuinn/obsidian-flashy/releases/tag/1.1.0

Please let me know if there's anything else needed. Thanks!

mnaoumov commented:
Please replace

```js
el.style.setProperty('--var-name', 'value');
```

with

```js
el.setCssProps({
  '--var-name': 'value'
});
```
